src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ExpressionTreeCallRewriter.cs

```diff
@@ -895,16 +895,11 @@ private object GetObject(Expr pExpr)
                     CType underlyingType = pExpr.Type;
                     object objval;
 
-                    if (pExpr.Type is NullType)
+                    if (underlyingType is NullType)
                     {
                         return null;
                     }
 
-                    if (pExpr.Type.isEnumType())
-                    {
-                        underlyingType = underlyingType.getAggregate().GetUnderlyingType();
-                    }
-
                     switch (Type.GetTypeCode(underlyingType.AssociatedSystemType))
                     {
```

Comment:
There's nowhere with this branch (there's one in `SymbolTable` that's almost the opposite, and one in `ConstVal` for comparing with the default for the type in question, but none for getting the value).

IMO it should ideally be in `ExprConstant` since that's where the type and `ConstVal` both are, and this whole method be a virtual property, but the implementation for some of the other `Expr` overrides would need a `TypeManager` value pushed into them on construction and it gets messy fast. However, next time I've no pending PRs (since it'll be a big change that will clash with everything) I'll make `TypeManager` static and that will get rid of that need, making changing this method to a virtual property easy.